### ISSUE TITLE
Launch Template with Managed Node Groups Example Incorrect Security Group

### ISSUE THREAD

**Issue by geoff-contino** [terraform-aws-modules/terraform-aws-eks#1662]
## Description

As per issue https://github.com/kubernetes-sigs/aws-efs-csi-driver/issues/574 it would appear that the example in module 

https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/examples/launch_templates_with_managed_node_groups/launchtemplate.tf

Attaches 

```
 module.eks_cluster.worker_security_group_id
```

to the launch template for the nodes - this appears to be incorrect and this should be:

```
module.eks_cluster.cluster_primary_security_group_id
```

Otherwise none of the EFS, ENIs can connect to the nodes because they use the cluster primary security group id as the reference.

Having checked the standard node group deployment, this appears to be the case...

**Comment by daroga0002** [comment]
> I believe `worker_security_group_id` should be mapped to https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_node_group#remote_access_security_group_id

 Nope, as this feature is working that AWS API is creating additional security group with open port 22 (SSH) and setup provided SG as source security group.

**Comment by daroga0002** [comment]
this is some bug which we must fix to have consistent behaviour between totally managed and "semi" managed node groups in EKS

**Comment by geoff-contino** [comment]
@daroga0002 - so it turns out we were setting the efs share to use that SG in a module within another module - it was not the default action. My fault - sorry!

That said - the cluster_sg should be applied to both the cluster and nodes by default (as it is when launched fully managed). This will also allow for this SG to be referenced and node groups to be destroyed and still allow resources access to the cluster.

**Comment by daroga0002** [comment]
seems be related to https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1616

**Comment by antonbabenko** [comment]
This issue has been resolved in version 18.0.0 :tada:
